experimental/InvariantTheory/primary_invariants.jl

```diff
@@ -28,6 +21,27 @@ function test_primary_degrees_via_hilbert_series(R::InvRing, degrees::Vector{Int
   return true
 end
 
+function reduce_hilbert_series_by_primary_degrees(R::InvRing)
+  fl, h = _reduce_hilbert_series_by_primary_degrees(R, [ total_degree(f.f) for f in primary_invariants(R) ])
+  @assert fl
+  return h
+end
+
+function _reduce_hilbert_series_by_primary_degrees(R::InvRing, degrees::Vector{Int})
+  mol = molien_series(R)
+  f = numerator(mol)
+  g = denominator(mol)
+  for d in degrees
+    # multiply f by 1 - t^d
+    f -= shift_left(f, d)
+  end
+  if !is_zero(mod(f, g))
+    return false, parent(f)()
+  end
+  h = div(f, g)
+  return true, h
```

Comment:
Isn't this the same as this?
```suggestion
  return divides(f, g)
```